omnibus/cookbooks/omnibus-supermarket/libraries/config.rb

```diff
@@ -58,21 +58,71 @@ def self.load_or_create_secrets!(filename, node)
+      all_s3_settings = required_s3_vars.all? { |key| !(config[key].nil? || config[key].empty?) }
 
       if any_s3_settings && !all_s3_settings
         raise IncompleteConfig.new "Got some, but not all, of the required S3 configs. Must provide none or all of #{required_s3_vars} to configure cookbook storage in an S3 bucket."
       end
 
-      if node['s3_bucket'] =~ /\./ &&
-         (node['s3_domain_style'] != ':s3_path_url' || node['s3_region'] != 'us-east-1')
+      if config['s3_bucket'] =~ /\./ &&
+         (config['s3_domain_style'] != ':s3_path_url' || config['s3_region'] != 'us-east-1')
          raise IncompatibleConfig.new "Incompatible S3 bucket settings. If the bucket name contains periods, the bucket must be in us-east-1 and the domain style must be :s3_path_url.\nAmazon recommends against periods in bucket names. See: https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html"
       end
     end
 
+    def self.audit_fips_config(config)
+      if !built_with_fips?(config['install_directory'])
+        if fips_enabled_in_kernel?
+          raise IncompatibleConfig.new "Detected FIPS is enabled in the kernel, but FIPS is not supported by this installer."
+        end
+        if config['fips_enabled']
+          raise IncompatibleConfig.new "You have enabled FIPS in your configuration, but FIPS is not supported by this installer."
+        end
+      end
+    end
+
+    def self.built_with_fips?(install_directory)
+      File.exist?("#{install_directory}/embedded/lib/fipscanister.o")
+    end
+
+    def self.fips_enabled_in_kernel?
+      fips_path = "/proc/sys/crypto/fips_enabled"
+      (File.exist?(fips_path) && File.read(fips_path).chomp != "0")
+    end
+
+    def self.maybe_turn_on_fips(node)
+      # the compexity of this method is currently needed to figure out what words to display
+      # to the poor human who has to deal with FIPS
+      case node['supermarket']['fips_enabled']
+      when nil
+        # the default value, set fips mode based on whether it is enabled in the kernel
+        node.normal['supermarket']['fips_enabled'] = Supermarket::Config.fips_enabled_in_kernel?
+        if node['supermarket']['fips_enabled']
+          Chef::Log.warn('Detected FIPS-enabled kernel; enabling FIPS 140-2 for Supermarket services.')
+        end
+      when false
+        node.normal['supermarket']['fips_enabled'] = Supermarket::Config.fips_enabled_in_kernel?
+        if node['supermarket']['fips_enabled']
+          Chef::Log.warn('Detected FIPS-enabled kernel; enabling FIPS 140-2 for Supermarket services.')
+          Chef::Log.warn('fips_enabled was set to false; ignoring this and setting to true or else Supermarket services will fail with crypto errors.')
+        end
+      when true
+        Chef::Log.warn('Overriding FIPS detection: FIPS 140-2 mode is ON.')
+      else
+        node.normal['supermarket']['fips_enabled'] = true
```

Comment:
Can we check for tihs condition in audit instead?

Reply:
This check-and-assignment is done here for the same reason as [explained on a different comment](https://github.com/chef/supermarket/pull/1768#discussion_r): this method `maybe_turn_on_fips` is about coercing `node['supermarket']['fips_enabled']` to a proper true/false boolean while `audit` is about raising errors for non-viable configurations.